core/src/HIP/Kokkos_HIP_Parallel_Team.hpp

```diff
@@ -673,14 +697,23 @@ class ParallelReduce<CombinedFunctorReducerType,
   __device__ inline void run(SHMEMReductionTag, int const threadid) const {
     const ReducerType& reducer = m_functor_reducer.get_reducer();
 
+    // FIXME_HIP comments below contain changes necessary to use word_size_type.
+    // Currently enabling them causes test failures and/or compiler errors
+    // elsewhere
+    /*integral_nonzero_constant<word_size_type, ReducerType::static_value_size()
+    / sizeof(word_size_type)> const word_count(reducer.value_size() /
+    sizeof(word_size_type));
+
+    reference_type value =
+        reducer.init(reinterpret_cast<pointer_type>(kokkos_impl_hip_shared_memory<word_size_type>()
+    + threadIdx.y * word_count.value));*/
```

Comment:
I don't see much point in keeping this code if it doesn't work. We have other pull requests that try to make small types work, see #5334.

Reply:
Agreed. I can remove the commented out code for getting smaller types to work (defining word_size_type and the changes you highlight here), but I think I should keep the FIXME_HIP for unconditionally setting UseShflReduction=false, as it is relevant and specific to this PR.